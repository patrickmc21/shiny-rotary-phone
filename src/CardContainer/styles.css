.Card-Container {
  border: 1px solid coral;
  margin: auto;
  width: 56vw;
}