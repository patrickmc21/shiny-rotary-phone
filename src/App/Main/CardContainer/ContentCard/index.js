import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const ContentCard = () => {
  return(
    <article className='content-card'>
      I'm the ContentCard
    </article>
  )
};

export default ContentCard;