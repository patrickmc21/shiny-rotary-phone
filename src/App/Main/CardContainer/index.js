import React from 'react';
import PropTypes from 'prop-types';
import ContentCard from './ContentCard/index.js';
import './styles.css';

const CardContainer = () => {
  const array = [];
  for (let i =0; i< 100; i++) {
    array.push(i);
  }

  const Cards = array.map(num => <ContentCard />)
  return(
    <section className='Card-Container'>
      {Cards}
    </section>
  )
}

export default CardContainer;