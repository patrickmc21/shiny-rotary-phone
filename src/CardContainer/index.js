import React from 'react';
import PropTypes from 'prop-types';
import ContentCard from '../ContentCard/index.js';
import './styles.css';

const CardContainer = ({activeCategoryInfo, activeCategoryName, addToFavorites, favorites}) => {
  const Cards = activeCategoryInfo.map(categoryItem => {
    const buttonClass = favorites.find(favorite => favorite.name === categoryItem.name) ? 'active' : '';
    return (
      <ContentCard 
        key={categoryItem.name}
        categoryItem={categoryItem}
        activeCategoryName={activeCategoryName}
        addToFavorites={addToFavorites}
        buttonClass={buttonClass}/>
    )
  })
  return(
    <section className='Card-Container'>
      <h2>{activeCategoryName}</h2>
      {Cards}
    </section>
  )
}

CardContainer.propTypes = {
  activeCategory: PropTypes.array,
  activeCategoryName: PropTypes.string,
  addToFavorites: PropTypes.func,
  favorites: PropTypes.array
};

export default CardContainer;