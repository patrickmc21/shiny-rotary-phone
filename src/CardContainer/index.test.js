import React from 'react';
import { shallow, mount } from 'enzyme';
import CardContainer from './index.js';

describe('CardContainer', () => {

  let wrapper;
  let mockCategoryInfo;
  let mockActiveCategoryName;
  let mockAddToFavorites;
  let mockFavorites;
  let mockUpdateCategory;
  let mockNext;
  let mockPrevious;

  beforeEach(() => {
    mockNext = 'next';
    mockPrevious = 'previous';
    mockActiveCategoryName = 'people';
    mockAddToFavorites = jest.fn();
    mockUpdateCategory = jest.fn();
    mockCategoryInfo = [
      {
        name: 'Han Solo',
        Homeworld: 'Corellia',
        'Homeworld Population': 10000,
        Species: 'human'
      },
      {
        name: 'Boba Fett',
        Homeworld: 'Kamino',
        'Homeworld Population': 100,
        Species: 'human'
      }
    ];
    mockFavorites = [
      {
        name: 'Han Solo',
        Homeworld: 'Corellia',
        'Homeworld Population': 10000,
        Species: 'human'
      }
    ];
    wrapper = shallow(
      <CardContainer 
        activeCategoryInfo={mockCategoryInfo}
        activeCategoryName={mockActiveCategoryName}
        addToFavorites={mockAddToFavorites}
        favorites={mockFavorites}
        updateCurrentCategory={mockUpdateCategory}
        next={mockNext}
        previous={mockPrevious}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should give a card an favorite class if content is in favorites', () => {
    const expected = 'favorite';
    expect(wrapper.find('ContentCard').first().props().buttonClass).toEqual(expected)
  });

  it('next button should run updateCurrentCategory with correct params', () => {
    wrapper.find('.next').simulate('click');
    expect(mockUpdateCategory).toHaveBeenCalledWith(mockActiveCategoryName, mockNext)
  });

  it('previous button should run updateCurrentCategory with correct params', () => {
    wrapper.find('.previous').simulate('click');
    expect(mockUpdateCategory).toHaveBeenCalledWith(mockActiveCategoryName, mockPrevious)
  });

})