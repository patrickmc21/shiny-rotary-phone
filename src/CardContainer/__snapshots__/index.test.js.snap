// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    activeCategoryInfo={
        Array [
            Object {
              "Homeworld": "Corellia",
              "Homeworld Population": 10000,
              "Species": "human",
              "name": "Han Solo",
            },
            Object {
              "Homeworld": "Kamino",
              "Homeworld Population": 100,
              "Species": "human",
              "name": "Boba Fett",
            },
          ]
    }
    activeCategoryName="people"
    addToFavorites={[Function]}
    favorites={
        Array [
            Object {
              "Homeworld": "Corellia",
              "Homeworld Population": 10000,
              "Species": "human",
              "name": "Han Solo",
            },
          ]
    }
    next="next"
    previous="previous"
    updateCurrentCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="next"
          onClick={[Function]}
>
          Next
</button>,
        <button
          className="previous"
          onClick={[Function]}
>
          Last
</button>,
        <h2>
          people
</h2>,
        <div
          className="card-wrapper"
>
          <ContentCard
                    activeCategoryName="people"
                    addToFavorites={[Function]}
                    buttonClass="favorite"
                    categoryItem={
                              Object {
                                        "Homeworld": "Corellia",
                                        "Homeworld Population": 10000,
                                        "Species": "human",
                                        "name": "Han Solo",
                                      }
                    }
          />
          <ContentCard
                    activeCategoryName="people"
                    addToFavorites={[Function]}
                    buttonClass=""
                    categoryItem={
                              Object {
                                        "Homeworld": "Kamino",
                                        "Homeworld Population": 100,
                                        "Species": "human",
                                        "name": "Boba Fett",
                                      }
                    }
          />
</div>,
      ],
      "className": "Card-Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "className": "next",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Last",
          "className": "previous",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Last",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "people",
        },
        "ref": null,
        "rendered": "people",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ContentCard
              activeCategoryName="people"
              addToFavorites={[Function]}
              buttonClass="favorite"
              categoryItem={
                            Object {
                                          "Homeworld": "Corellia",
                                          "Homeworld Population": 10000,
                                          "Species": "human",
                                          "name": "Han Solo",
                                        }
              }
/>,
            <ContentCard
              activeCategoryName="people"
              addToFavorites={[Function]}
              buttonClass=""
              categoryItem={
                            Object {
                                          "Homeworld": "Kamino",
                                          "Homeworld Population": 100,
                                          "Species": "human",
                                          "name": "Boba Fett",
                                        }
              }
/>,
          ],
          "className": "card-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Han Solo",
            "nodeType": "function",
            "props": Object {
              "activeCategoryName": "people",
              "addToFavorites": [Function],
              "buttonClass": "favorite",
              "categoryItem": Object {
                "Homeworld": "Corellia",
                "Homeworld Population": 10000,
                "Species": "human",
                "name": "Han Solo",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Boba Fett",
            "nodeType": "function",
            "props": Object {
              "activeCategoryName": "people",
              "addToFavorites": [Function],
              "buttonClass": "",
              "categoryItem": Object {
                "Homeworld": "Kamino",
                "Homeworld Population": 100,
                "Species": "human",
                "name": "Boba Fett",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="next"
            onClick={[Function]}
>
            Next
</button>,
          <button
            className="previous"
            onClick={[Function]}
>
            Last
</button>,
          <h2>
            people
</h2>,
          <div
            className="card-wrapper"
>
            <ContentCard
                        activeCategoryName="people"
                        addToFavorites={[Function]}
                        buttonClass="favorite"
                        categoryItem={
                                    Object {
                                                "Homeworld": "Corellia",
                                                "Homeworld Population": 10000,
                                                "Species": "human",
                                                "name": "Han Solo",
                                              }
                        }
            />
            <ContentCard
                        activeCategoryName="people"
                        addToFavorites={[Function]}
                        buttonClass=""
                        categoryItem={
                                    Object {
                                                "Homeworld": "Kamino",
                                                "Homeworld Population": 100,
                                                "Species": "human",
                                                "name": "Boba Fett",
                                              }
                        }
            />
</div>,
        ],
        "className": "Card-Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "className": "next",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Last",
            "className": "previous",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Last",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "people",
          },
          "ref": null,
          "rendered": "people",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ContentCard
                activeCategoryName="people"
                addToFavorites={[Function]}
                buttonClass="favorite"
                categoryItem={
                                Object {
                                                "Homeworld": "Corellia",
                                                "Homeworld Population": 10000,
                                                "Species": "human",
                                                "name": "Han Solo",
                                              }
                }
/>,
              <ContentCard
                activeCategoryName="people"
                addToFavorites={[Function]}
                buttonClass=""
                categoryItem={
                                Object {
                                                "Homeworld": "Kamino",
                                                "Homeworld Population": 100,
                                                "Species": "human",
                                                "name": "Boba Fett",
                                              }
                }
/>,
            ],
            "className": "card-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Han Solo",
              "nodeType": "function",
              "props": Object {
                "activeCategoryName": "people",
                "addToFavorites": [Function],
                "buttonClass": "favorite",
                "categoryItem": Object {
                  "Homeworld": "Corellia",
                  "Homeworld Population": 10000,
                  "Species": "human",
                  "name": "Han Solo",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Boba Fett",
              "nodeType": "function",
              "props": Object {
                "activeCategoryName": "people",
                "addToFavorites": [Function],
                "buttonClass": "",
                "categoryItem": Object {
                  "Homeworld": "Kamino",
                  "Homeworld Population": 100,
                  "Species": "human",
                  "name": "Boba Fett",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
